core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ResourceSearchRequest.java

```diff
@@ -66,5 +74,22 @@ protected Resources toCollectionResource(RepositoryContext context, Hits<Resourc
 	protected Resources createEmptyResult(int limit) {
 		return new Resources(limit, 0);
 	}
-
+	
+	@Override
+	protected void toQuerySortBy(RepositoryContext context, Builder sortBuilder, Sort sort) {
+		if (sort instanceof SortField) {
+			SortField sortField = (SortField) sort;
+			if (ResourceSearchRequestBuilder.TYPE_SORT.equals(sortField.getField())) {
+				HashMap<String, String> orderMap = Maps.newHashMap();
+				Registry registry = context.service(ResourceTypeConverter.Registry.class);
+				registry.getResourceTypeConverters().values().forEach(converter -> {
+					orderMap.put(converter.getResourceType(), converter.getRank());
+				});
+				sortBuilder.sortByScript(ResourceSearchRequestBuilder.TYPE_SORT, Map.of("orderByType", orderMap) , sort.isAscending() ? Order.ASC : Order.DESC);
```

Comment:
Rename script parameter `orderByType` to `ranks`.

Reply:
Done